### DISCUSSION TITLE
master layout stack: change to last in first out?

### DISCUSSION THREAD

**Discussion by setseed13** [hyprwm/Hyprland#6827] [General]
I'm not sure whether to file this as an issue, Q&A or something else, so I just put this into General, I hope this is ok.
Additionally, yes, I searched the wiki (extensively), but then again I easily miss stuff, so I apologize in advance in case that some option eluded me.

I'm using the master layout, and for me the stack behaves non-intuetively.
Let's say I open three windows `A`, `B` and `C` in this sequence. I end up with

```
 ___   ___
|   | | A |
| C | |___|
|   | | B | 
|___| |___|
```
Now I close `C` and I get

```
 ___   ___
|   | |   |
| A | | B |
|   | |   | 
|___| |___|
```
and `A` is the new master.
However, I would expect it be reversed, that is last in, first out.
Not only because it makes more sense in my head, but also because this breaks my work flow routinely.
I work on something in `B`, spawn a terminal  `C` and 2 commands later I'm rearranging my layout...

So, any chance this will somehow be implemented, eg
```
master {
    stack = {last_out, first_out}
}
```

Thanks!

**Comment by jkerdreux-imt** [comment]
There is no way to do this right now I think. The last master can be master only if you change the stack order. This will change the display order (which bother me too), but the behavior will be ok. You will get this: 

```
 ___   ___
|   | | B |
| C | |___|
|   | | A | 
|___| |___|

```

To achieve this, just set the new_on_top

```
master {
    new_on_top = true
}
```

Hope this helps

**Comment by setseed13** [comment]
thank you so much, this is exactly what I wanted (dont care about the sstack itself)
I never tried it, because I totally misread that option
